### ISSUE TITLE
LGBMRanker unexpected predictions

### ISSUE THREAD

**Issue by dirknbr** [microsoft/LightGBM#2968]
I find some strange behaviour in the LGBMRanker, in the below example my label is a rank from 0 to 19 and one of the features is the label. The prediction function returns all 0, why? I would expect near perfect fit.

also posted here https://stackoverflow.com/questions/60995634/lightgbm-ranker-predictions-are-all-0

py: 3.7

```
import numpy as np
from lightgbm import LGBMRanker

N = 20
y = np.arange(N)
X = np.random.normal(size=(N, 2))
X[:, 1] = y[0:]

model = LGBMRanker()
model.fit(X, y, group=[N])
print(model.predict(X))
```
-> [0, 0,..., 0]

**Comment by guolinke** [comment]
please try `min_data=1` and `min_data_in_bin=1` in your parameter.

**Comment by dirknbr** [comment]
thanks it works now
